rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetTelemetryNodeTest.java

```diff
@@ -15,72 +15,341 @@
+            assertThat(node.parseAggregationConfig(aggEnum.name())).isEqualTo(aggEnum);
         }
     }
 
     @Test
-    public void givenAggregationWhiteSpace_whenParseAggregation_thenException() {
-        Assertions.assertThrows(IllegalArgumentException.class, () -> {
-            node.parseAggregationConfig(" ");
-        });
+    public void givenAggregationWhiteSpace_whenParseAggregation_thenException() throws TbNodeException {
+        config.setFetchMode("ALL");
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+        assertThatThrownBy(() -> node.parseAggregationConfig(" ")).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void givenAggregationIncorrect_whenParseAggregation_thenException() throws TbNodeException {
+        config.setFetchMode("ALL");
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+        assertThatThrownBy(() -> node.parseAggregationConfig("TOP")).isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
-    public void givenAggregationIncorrect_whenParseAggregation_thenException() {
-        Assertions.assertThrows(IllegalArgumentException.class, () -> {
-            node.parseAggregationConfig("TOP");
-        });
+    public void verifyDefaultConfig() {
+        assertThat(config.getStartInterval()).isEqualTo(2);
+        assertThat(config.getEndInterval()).isEqualTo(1);
+        assertThat(config.getStartIntervalPattern()).isEqualTo("");
+        assertThat(config.getEndIntervalPattern()).isEqualTo("");
+        assertThat(config.isUseMetadataIntervalPatterns()).isFalse();
+        assertThat(config.getStartIntervalTimeUnit()).isEqualTo(TimeUnit.MINUTES.name());
+        assertThat(config.getEndIntervalTimeUnit()).isEqualTo(TimeUnit.MINUTES.name());
+        assertThat(config.getFetchMode()).isEqualTo("FIRST");
+        assertThat(config.getOrderBy()).isEqualTo("ASC");
+        assertThat(config.getAggregation()).isEqualTo(Aggregation.NONE.name());
+        assertThat(config.getLimit()).isEqualTo(1000);
+        assertThat(config.getLatestTsKeyNames()).isEmpty();
     }
 
+    @ParameterizedTest
+    @MethodSource
+    public void givenFetchModeAndLimit_whenInit_thenVerifyLimit(String fetchMode, int limit, int expectedLimit) throws TbNodeException {
+        config.setFetchMode(fetchMode);
+        config.setLimit(limit);
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        var actualLimit = ReflectionTestUtils.getField(node, "limit");
+        assertThat(actualLimit).isEqualTo(expectedLimit);
+    }
+
+    private static Stream<Arguments> givenFetchModeAndLimit_whenInit_thenVerifyLimit() {
+        return Stream.of(
+                Arguments.of("FIRST", 0, 1),
+                Arguments.of("LAST", 10, 1),
+                Arguments.of("ALL", 0, 1000),
+                Arguments.of("ALL", 5, 5)
+        );
+    }
```

Comment:
BE validation of limit does not cover all cases. It is possible to edit rule chain JSON and break the validation. Please ensure that BE and FE validation is in sync